### DISCUSSION TITLE
Control vector parameterization in CasADi

### DISCUSSION THREAD

**Discussion by PrakitrSrisuma** [casadi/casadi#3087] [Q&A]
Is it possible to do piecewise linear control vector parameterization (instead of piecewise constant) in CasADi, e.g., using the example direct_multiple_shooting.m?

**Comment by jaeandersson** [comment]
Yes, just add another state $x_\text{new}$ defined by:

$$\dot{x}_{\text{new}}(t) = u(t)$$

You can let $x_\text{new}(0)$ be free in the optimal control formulation.

**Reply by PrakitrSrisuma** [reply]
Hello, thank you for your reply. I tried that but it did not work. Could you elaborate more why the above formulation would give a piecewise linear control vector?

**Reply by jaeandersson** [reply]
If the control vector is piecewise linear, then its derivative with respect to time is piecewise constant.
